docs/concepts/vscode-extension.md

```diff
@@ -13,126 +15,140 @@ It utilizes the excellent [demisto-sdk](./demisto-sdk) python package.
-{
-    "version": "0.2.0",
-    "configurations": [
-        {
-            "name": "Python: Run Custom Integration",
-            "type": "python",
-            "request": "launch",
-            "program": "${file}",  // File example: Packs/Pasta/Integrations/Pasta/Pasta.py
-            "console": "integratedTerminal",
-            "cwd": "${fileDirname}" // The directory name of the Pasta integration (Packs/Pasta/Integrations/Pasta/).
-        }
-    ]
-}
-```
+## Notes
+
+* If during the installation one or more Python packages fail to install, the installation will proceed and create the virtual environment with the packages that were installed.
+
+  It is your responsibility to figure out why the installation failed (probably due to missing non-Python dependencies in your host), and manually install or recreate the virtual environnement afterwards.
+  If you face this issue, you might want to try [Open integration/script in Dev Container](#open-integrations-and-scripts-in-dev-container-advanced) option.
+* To use [demisto-sdk](./demisto-sdk) in this workspace it has do be installed globally in your system, with `pip install demisto-sdk` in the system interpreter.
+* You can use this feature together with the content Dev Container.
+
+## Open integrations and scripts in Dev Container (Advanced)
+
+The `XSOAR` integrations and scripts actually run on a dedicated Docker image.
+In some integrations, the Python environment is not enough, and it is not easy to install the system dependencies.
 
-Now you can run your custom integration with your pipenv!
+With this feature, you are able to develop inside the integration or a script Docker image, with **all** dependencies preinstalled.
+This will use the Docker image that is in the `YAML` file.
+
+### System requirements
+
+**Windows**: Docker Desktop 2.0+ on Windows 10 Pro/Enterprise. Windows 10 Home (2004+) requires Docker Desktop 2.3+ and the WSL 2 back-end.
+**macOS**: Docker Desktop 2.0+.
+**Linux**: Docker CE/EE 18.06+ and Docker Compose 1.21+.
+
+### Installation
+
+Follow the [VSCode documentation](https://code.visualstudio.com/docs/remote/containers#_installation).
+
+### Usage
+
+* Go to the integration or a script.
+* Right-click it and select **Open integration/script in a Dev Container**
+
+### Features
+
+The same features of the [virtual environment](#open-integrations-and-scripts-in-python-virtual-environment) feature is available here. 
+
+
+## Troubleshooting
+
+- If **Open integration/script in a Dev Container** or **Open integration/script in a virtual environment** fails, make sure the **Docker** is running. In addition, you can try to [clean up the Docker](https://docs.docker.com/config/pruning/) or [sign in to docker](https://www.docker.com/blog/seamless-sign-in-with-docker-desktop-4-4-2/) to avoid the [docker pull rate limit](https://docs.docker.com/docker-hub/download-rate-limit/#:~:text=Pull%20rates%20limits%20are%20based,to%205000%20pulls%20per%20day.).
```

Comment:
```suggestion
### I**Open integration/script in a Dev Container** or **Open integration/script in a virtual environment** fails
1. Make sure **Docker** is running.
2. If *Docker* is running try to [clean up the Docker](https://docs.docker.com/config/pruning/) or [sign in to docker](https://www.docker.com/blog/seamless-sign-in-with-docker-desktop-4-4-2/) to avoid the [docker pull rate limit](https://docs.docker.com/docker-hub/download-rate-limit/#:~:text=Pull%20rates%20limits%20are%20based,to%205000%20pulls%20per%20day.).
```